### DISCUSSION TITLE
Full coupled electrochemical-thermal model?

### DISCUSSION THREAD

**Discussion by thuwangming** [pybamm-team/PyBaMM#1868] [Q&A] [answered]
Hello,
I have a question that whether PyBamm has implemented the full coupled electrochemical-thermal model?
I mean that the "Cell Temperature"  or "X-averaged Cell Temperature" variable can be calculated at each time step, and meanwhile, the parameters, e.g. electrolyte diffusivity, conductivity, transference number, will also be updated with respect to the updated temperature. I noticed the get_coupled_variables function in the electrolyte submodels, but sorry I have not read the full source codes of Pybamm, so I don't confirm how the solver runs.
Thanks a lot!

**Comment by brosaplanella** [comment, marked as answer]
Hi @thuwangming! This [notebook](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/models/thermal-models.ipynb) explains the different thermal models available in PyBaMM.

**Reply by rtimms** [reply]
and, just to add, the parameters can depend on the temperature. E.g. see [here](https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/input/parameters/lithium_ion/negative_electrodes/graphite_Ecker2015/graphite_diffusivity_Ecker2015.py).

**Comment by thuwangming** [comment]
Thanks!  @brosaplanella @rtimms , I will take a look.
